### PR TITLE
Fixes #47538: Add functionality for manually creating a Job instance from a CronJob

### LINKED ISSUE #47538: Ability to start CronJobs manually
It would be nice to support manual runs for CronJobs. Today, IIRC, a CronJob misses its schedule for whatever reason and we won't start it until the next time it's scheduled to run. At the very least, this could be a kubectl generator, eventually we may want to support it as part of the CJ API.

@stevekuznetsov since you were asking about it today

@kubernetes/sig-apps-feature-requests

### PR DESCRIPTION
**Pull request by erhudy**
**What this PR does / why we need it**: This changeset adds the command `kubectl create job` with the flag `--from-cronjob`, which allows a user to create a Job from a CronJob via the CLI.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Fixes #47538

**Special notes for your reviewer**: for @soltysh, who has been guiding me through getting this PR up

**Release note**:

```release-note
* adds `kubectl create job --from-cronjob` command for creating a Job from an existing CronJob
```

### CODE REVIEW

File: pkg/kubectl/cmd/create_job.go
Comment (soltysh) [inline pkg/kubectl/cmd/create_job.go]: No need to get the job twice, here and in run.

File: pkg/kubectl/cmd/create_job.go
Comment (soltysh) [inline pkg/kubectl/cmd/create_job.go]: Drop it.

File: pkg/kubectl/cmd/create_job.go
Comment (soltysh) [inline pkg/kubectl/cmd/create_job.go]: Put a comment here, that the job name can't be longer than `DNS1035LabelMaxLength` which is 52 chars.

File: pkg/kubectl/cmd/create_job.go
Comment (soltysh) [inline pkg/kubectl/cmd/create_job.go]: IIRC we wanted to copy the labels from the template, similarly how the controller does it in [here](https://github.com/kubernetes/kubernetes/blob/89ac18de1a71d08d2bd7654c325cd8aa476190bd/pkg/controller/cronjob/utils.go#L177-L178).

File: pkg/kubectl/cmd/create_job.go
Comment (soltysh) [inline pkg/kubectl/cmd/create_job.go]: Drop it.

**Review by soltysh — APPROVED** [review]

**Review by soltysh — CHANGES_REQUESTED** [review]
I left you some comments.

File: pkg/kubectl/cmd/create_job.go
Comment (soltysh) [inline pkg/kubectl/cmd/create_job.go]: Drop it.

File: pkg/kubectl/cmd/create_job.go
Comment (soltysh) [inline pkg/kubectl/cmd/create_job.go]: Drop entirely the `getCronJob` method.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Review by soltysh — APPROVED** [review]
/lgtm

**Review by soltysh — APPROVED** [review]
/lgtm

File: pkg/kubectl/cmd/create_job.go
Comment (deads2k) [inline pkg/kubectl/cmd/create_job.go]: `--from` should accept a resource style string (like `kubectl get`) and you can get the resource.group/name from there

**Comment by deads2k** [comment]
In concept fine.  I'd like to see the `--from` refined to take a resource string.

/approve

**Comment by soltysh** [comment]
> In concept fine. I'd like to see the --from refined to take a resource string.

I can fix that in the followup where I'll be adding functionality of creating regular jobs.

File: pkg/kubectl/cmd/create_job_test.go
Comment (droot) [inline pkg/kubectl/cmd/create_job_test.go]: might want to define a const for "cronjob.kubernetes.io/instantiate" because I see these popping up quite a few times in the code.

**Review by droot — APPROVED** [review]
/lgtm

File: pkg/kubectl/cmd/create_job_test.go
Comment (soltysh) [reply pkg/kubectl/cmd/create_job_test.go]: I'm not sure I'm wanna keep it, since it doesn't do us anything.

**Comment by soltysh** [comment]
/hold
@erhudy would you mind rebasing your PR the `PrintObject` has a different signature now.

**Comment by erhudy** [comment]
@soltysh rebased and fixed the PrintObject() signature.

**Review by soltysh — APPROVED** [review]
/lgtm
/hold cancel

**Comment by mikedanese** [comment]
I want to get this in so we can unblock batch merges. It's currently interacting poorly with #60117 and it's higher in the queue so I'm giving it priority. @deads2k will have to rebase.

**Comment by deads2k** [comment]
> In concept fine. I'd like to see the --from refined to take a resource string.
> I can fix that in the followup where I'll be adding functionality of creating regular jobs.

You can't really.  This is adding a flag (API) we don't want.

/hold

Holding on a flag problem.

**Comment by erhudy** [comment]
Closing this in favor of #60084.

**Pull request closed without merging**
